src/DotLiquid.Tests/StandardFilterTests.cs

```diff
@@ -438,7 +478,7 @@ public void TestMap()
                         Hash.FromAnonymousObject(new { foo = Hash.FromAnonymousObject(new { bar = "c" }) })
                     }
                     }));
-            CollectionAssert.AreEqual(new[] { new { a = 1 }, new { a = 2 }, new { a = 3 }, new { a = 4 } },
+            CollectionAssert.AreEqual(new object[] { null, null, null, null },

```

Comment:
This is a breaking change, albeit understandable. More on the actual line that causes this change.

Reply:
There is this explicit change and the removed tests above. Regardless of whether the new behavior is right or wrong, this is indeed an explicit breaking change. As such, at minimum the change must be behind an opt-in version compatibility flag, so we don't risk breaking consuming projects.

Reply:
This test was supposed to map property "b" from the array of objects, but these objects don't have property b. The previous result is pretty bizarre. I struggle to find the value in that past behavior.

Reply:
While normally I do say that we have to provide compatibility with explicitly tested behavior, behaviors we have knowledge a consuming project is relying on, or behavior one could reasonably even if incorrectly expect, I don't understand how the past behavior of preserving the array made sense. Could it be that some consumers applied multiple maps in a chain, expecting the first matching property to be rendered??